packages/npm-dependencies-validation/src/utils/packageJsonUtil.ts

```diff
@@ -0,0 +1,116 @@
+import { access, readFile } from "fs/promises";
+import { constants } from "fs";
+import { join, dirname, sep, relative } from "path";
+import { VscodeFsUri, VscodeWsFolder } from "../types";
+
+const yarnManagerFiles = [
+  "yarn.lock",
+  ".yarnrc",
+  ".yarnrc.yml",
+  ".yarn",
+  "./node_modules/.yarn-integrity",
+];
+const pnpmManagerFiles = [
+  "pnpm-workspace.yaml",
+  "pnpm-lock.yaml",
+  ".pnpmfile.cjs",
+  "./node_modules/.pnpm",
+];
+const npm7Properties = ["workspaces"];
+
+async function readJsonFile(packageJsonUri: VscodeFsUri): Promise<any> {
+  try {
+    const packageJsonContent = await readFile(packageJsonUri.fsPath, "utf-8");
+    const content: { name: string } = JSON.parse(packageJsonContent);
+    return content;
+  } catch (error: any) {
+    return {};
+  }
+}
+
+async function isPathExist(packageJsonPath: string): Promise<boolean> {
+  try {
+    await access(packageJsonPath, constants.R_OK | constants.W_OK);
```

Comment:
access checks the permissions, in theory our process may not have permissions
even though the file exists.
- https://nodejs.org/api/fs.html#fspromisesaccesspath-mode

Reply:
I think you should use one of the fs.stat variants

Reply:
Using fs.stat() to check for the existence of a file before calling fs.open(), fs.readFile() or fs.writeFile() is not recommended. Instead, user code should open/read/write the file directly and handle the error raised if the file is not available.

To check if a file exists without manipulating it afterwards, fs.access() is recommended.